src/Microsoft.Identity.Core/Platforms/android/LegacyCachePersistance.cs

```diff
@@ -28,66 +28,54 @@
 using Android.App;
 using Android.Content;
 using System;
+using Microsoft.Identity.Core;
 
-namespace Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Cache
+namespace Microsoft.Identity.Core.Cache
 {
     [Android.Runtime.Preserve(AllMembers = true)]
-    internal class TokenCachePlugin
+    internal class LegacyCachePersistance
     {
         private const string SharedPreferencesName = "ActiveDirectoryAuthenticationLibrary";
         private const string SharedPreferencesKey = "cache";
         
-        public static void BeforeAccess(TokenCacheNotificationArgs args)
+        public static byte[] LoadCache()
         {
-            if (args.TokenCache.Count > 0)
-            {
-                // We assume that the cache has not changed since last write
-                return;
-            }
-
             try
             {
                 ISharedPreferences preferences = Application.Context.GetSharedPreferences(SharedPreferencesName, FileCreationMode.Private);
                 string stateString = preferences.GetString(SharedPreferencesKey, null);
                 if (stateString != null)
                 {
-                    byte[] state = Convert.FromBase64String(stateString);
-                    args.TokenCache.Deserialize(state);
+                    return Convert.FromBase64String(stateString);
                 }
             }
             catch (Exception ex)
             {
-                CallState.Default.Logger.Warning(null, "Failed to load cache: ");
-                CallState.Default.Logger.ErrorPii(null, ex);
+                CoreLoggerBase.Default.Warning("Failed to load cache: " + ex.Message);
```

Comment:
I noticed several places where we don't double up the log messages for pii and no-pii. I thought we agreed to double up on messages so something that would be logged just to no-pii would also get logged in pii logs for log purposes on our end. Otherwise, if dev has piiEnabled, we will only get a small amount of log messages. I can make the changes once you merge. Also, in this case, why is one log warning and the other errorPii?

Reply:
I kept the message level as is. I agree some logs updates are needed but they are orthogonal to the change here. Also, I think errorPii is used because it logs the exception object, but still inconsistent.

Reply:
Ok....that's why I approved the PR because I knew these changes weren't necessary. Since your changes impacted all the logging functionality, it was easy to see.  I will update the logs.

Reply:
@jennyf19 let's raise a bug for the missing log messages.